psycopg/setup.py

```diff
@@ -47,6 +47,7 @@
     # Requirements needed for development
     "dev": [
         "black",
+        "dnspython ~= 2.1.0",
```

Comment:
I think this would make more difficult than expected to set up the test environment. Not so much for this module, but shapely might need some external dependencies to come up. Honestly it's a direction I'd prefer not to go.

Reply:
I think shapely would not need to be added here because it is not typed (in contrast with dnspython) and thus already has an override defined in  pyproject.toml. Are we fine then? Alternatively, I can add this to `tox.ini` only in the `mypy` environment.

Reply:
Not entirely sold on it, but ok...